src/mlpack/methods/ann/ffn.hpp

```diff
@@ -142,6 +142,16 @@ class FFN
   void Predict(arma::mat& predictors, arma::mat& results);
 
   /**
+   * Predict the responses to a given set of predictors. The responses will
+   * reflect the output of the given output layer as returned by the
+   * output layer function.
+   *
+   * @param predictors Input predictors.
+   * @return Output predictions of responses into.
+   */
+  arma::mat Predict(arma::mat predictors);
+
```

Comment:
If `void Predict(const mat& predictors, mat& output);` and `void Predict(mat& predictors, mat& output);` exist together, I think it will be somehow confusing. In that case, assume we have a non-const `predictors`, then the function call will be dispatched to `void Predict(mat& predictors, mat& output);` but actually we may still want the `predictors` to stayed unchanged even we didn't mark it as const. So I think if we use these two overloads, we should delete the original one or move it to private. Furthure more, I don't like the idea to putut `output` as a parameter. It's very user-unfriendly -- almost every time we call the function we need to write an extra line. So what I propose is deleting the original one and adding `mat Predict(const mat&)` and `mat Predict(mat&&)`. If both of you agree, I can do this for ffn and rnn. Can I assume there would not be many regression issues due to this change?